volttron/utils/rmq_setup.py

```diff
@@ -488,120 +508,93 @@ def _verify_and_save_instance_ca(instance_ca_path, instance_ca_key):
         # openssl crl2pkcs7 -nocrl -certfile volttron2-ca.crt | openssl pkcs7 -print_certs  -noout
         # this should list subject, issuer of both root CA and
         # intermediate CA
-        crts.save_cert(instance_ca_path)
-        crts.save_key(instance_ca_key)
+        rmq_config.crts.save_cert(instance_ca_path)
+        rmq_config.crts.save_key(instance_ca_key)
     return found
 
 
-def setup_rabbitmq_volttron(type, verbose=False, prompt=False):
+def setup_rabbitmq_volttron(setup_type, verbose=False, prompt=False):
     """
     Setup VOLTTRON instance to run with RabbitMQ message bus.
-    :param type:
+    :param setup_type:
             single - Setup to run as single instance
             federation - Setup to connect multiple VOLTTRON instances as
                          a federation
             shovel - Setup shovels to forward local messages to remote instances
     :return:
     """
-    global instance_name, _log, crts, volttron_home, volttron_rmq_config
-
-    # Initialize global variable. Do it here as this method might be called
-    # from volttron-cfg. volttron-cfg might set vhome
+    instance_name = get_platform_instance_name(prompt=True)
+    # Store config this is checked at startup
+    store_message_bus_config(message_bus='rmq', instance_name=instance_name)
 
-    volttron_home = get_home()
-    volttron_rmq_config = os.path.join(volttron_home, 'rabbitmq_config.yml')
+    rmq_config = RMQConfigParams()
     if verbose:
         _log.setLevel(logging.DEBUG)
         _log.debug("verbose set to True")
         _log.debug(get_home())
 
-    instance_name = get_platform_instance_name(prompt=True)
-    # Store config this is checked at startup
-    store_message_bus_config(message_bus='rmq', instance_name=instance_name)
-
-    crts = certs.Certs()
+    rmq_config.crts = certs.Certs()
```

Comment:
This line is not necessary. Also RabbitmqMgmt class already has an instance of rmqconfig. So everything can use that. The only place I see we have a requirement to instantiate RabbitmqConfig class is when we do a create_rabbitmq_config. Others can simply use the RabbitmqMgmt class and the config within it.

Reply:
Done